lib/Runtime/Types/ScriptFunctionType.cpp

```diff
@@ -31,7 +31,7 @@ namespace Js
             scriptContext, functionPrototype,
             address,
             proxy->GetDefaultEntryPointInfo(),
-            library->ScriptFunctionTypeHandler(proxy->IsLambda() || proxy->IsAsync() || proxy->IsClassMethod(), proxy->GetIsAnonymousFunction()),
+            library->ScriptFunctionTypeHandler(proxy->IsLambda() || !proxy->IsConstructor() || proxy->IsAsync() || proxy->IsClassMethod(), proxy->GetIsAnonymousFunction()),
```

Comment:
`IsConstructor` implies `!(IsLambda || IsAsync || IsClassMethod)` so you can remove the checks for `IsLambda` etc in favor of just checking `IsConstructor`